### DISCUSSION TITLE
Where to find desktop Python BLE Central example

### DISCUSSION THREAD

**Discussion by penitko** [micropython/micropython#13756] [General]
Micropython has published many good BLE (gatt server, periperal ) examples like  ble_uart_peripheral.py here:
https://github.com/micropython/micropython/tree/master/examples/bluetooth

For that peripheral example, there is also a gatt client (central) example ble_simple_central.py.

But I would also be interested in a central Python example that would work in a Linux BlueZ desktop environment (e.g Ubuntu), which would communicate with the ble_uart_peripheral.py example. Where would one find such an example?

**Comment by dlech** [comment]
https://github.com/hbldh/bleak/tree/master/examples

**Reply by penitko** [reply]
It has never before happened to me that two example applications work immediately with each other completely unchanged.
It must have been the first time in my history as a software hobbyist.
So these aplications work in pairs: ble_simple.peripheral.py and uart_service.py
My devices are Ubuntu 22.04 with Bluetooth Dongle as Central and a ESP32 as BLE peripheral.

![ubuntu-central-and-esp32-peripheral](https://github.com/micropython/micropython/assets/98764466/4134f9eb-1e44-4a69-8e07-4cf7de734f96)
